### PR TITLE
Resolve memory unsafety in unpacking C enum

### PR DESCRIPTION
**Pull request by kbarros**
Intended to resolve https://github.com/singularitti/Spglib.jl/issues/183. I'm happy to iterate with you, @singularitti, on the code style or comments.

### CODE REVIEW

File: src/error.jl [hunk 1]
```diff
@@ -1,41 +1,42 @@
...
+const SPGLIB_SUCCESS = 0
+const SPGERR_SPACEGROUP_SEARCH_FAILED = 1
+const SPGERR_CELL_STANDARDIZATION_FAILED = 2
+const SPGERR_SYMMETRY_OPERATION_SEARCH_FAILED = 3
+const SPGERR_ATOMS_TOO_CLOSE = 4
+const SPGERR_POINTGROUP_NOT_FOUND = 5
+const SPGERR_NIGGLI_FAILED = 6
+const SPGERR_DELAUNAY_FAILED = 7
+const SPGERR_ARRAY_SIZE_SHORTAGE = 8
+const SPGERR_NONE = 9
```
Comment (kbarros) [inline src/error.jl:13]: These error codes (1 .. 9) aren't currently used. Removing them from the PR might make sense, at your disgression.

File: src/error.jl [hunk 2]
```diff
@@ -1,41 +1,42 @@
...
     msg::String
 end
 
+# The return value for the C function `spg_get_error_code` is a C enum called
+# `SpglibError`. The C compiler guarantees that enums are internally stored as
+# type `int`, which Julia represents as `Cint`. The size of `int` is
+# "implementation dependent". Frequently 4 or 8 bytes, but this depends on
+# architecture and compiler details: https://stackoverflow.com/a/366033/500314.
+# Directly casting this Cint into an @enum or SumTypes.jl value would be memory
+# unsafe (see https://github.com/singularitti/Spglib.jl/issues/183).
```
Comment (kbarros) [inline src/error.jl:25]: Feel free to delete this comment prior to merging if you prefer.

File: src/error.jl [hunk 3]
```diff
@@ -1,41 +1,42 @@
...
 """
     get_error_code()
 
 Return an instance of the enumerated type `SpglibReturnCode`.
 """
-get_error_code() = @ccall libsymspg.spg_get_error_code()::SpglibReturnCode
+get_error_code() = @ccall libsymspg.spg_get_error_code()::Cint
```
Comment (kbarros) [inline src/error.jl:31]: The return type of `Cint` is critical here -- that's what `spg_get_error_code` actually produces.

File: src/error.jl [hunk 4]
```diff
@@ -1,41 +1,42 @@
...
 
 """
     get_error_message(code::SpglibReturnCode)
 
 Get the corresponding error message for a given error code.
 """
-get_error_message(code::SpglibReturnCode) =
-    unsafe_string(@ccall libsymspg.spg_get_error_message(code::SpglibReturnCode)::Cstring)
+get_error_message(code::Cint) =
+    unsafe_string(@ccall libsymspg.spg_get_error_message(code::Cint)::Cstring)
```
Comment (kbarros) [inline src/error.jl:39]: This `code` parameter must also be of type `Cint`.

File: src/error.jl (hunk 2 above)
Comment (singularitti) [reply src/error.jl:25]: I like this!

**Pull request merged**